src/fileio.c

```diff
@@ -2193,27 +2193,6 @@ file_name_case_insensitive_p (const char *filename)
 #endif
 }
 
-DEFUN ("file-name-case-insensitive-p", Ffile_name_case_insensitive_p,
-       Sfile_name_case_insensitive_p, 1, 1, 0,
-       doc: /* Return t if file FILENAME is on a case-insensitive filesystem.
-The arg must be a string.  */)
-  (Lisp_Object filename)
-{
-  Lisp_Object handler;
-
-  CHECK_STRING (filename);
-  filename = Fexpand_file_name (filename, Qnil);
-
-  /* If the file name has special constructs in it,
-     call the corresponding file handler.  */
-  handler = Ffind_file_name_handler (filename, Qfile_name_case_insensitive_p);
-  if (!NILP (handler))
-    return call2 (handler, Qfile_name_case_insensitive_p, filename);
-
-  filename = ENCODE_FILE (filename);
```

Comment:
This is your mistake in the Rust. You are recursively calling the Rust function instead of the C function.

This means `file-name-case-insensitive-p` needs a `file_name_case_insensitive_p_rust` version and name in the `lisp_fn`.

Reply:
Good catch! I had no idea what the problem was.

Reply:
I missed it the first time too.

I ran temacs in lldb. It would segfault so then I looked at the backtrace. There was a LONG list of the same function when it should not be recursing.